// @arduino/arduino-iot-client/dist/model/ArduinoLinkedvariable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoTimezone = _interopRequireDefault(require("./ArduinoTimezone"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Arduino IoT Cloud API
                                                                                                                                                                                                                                                                                                                                                                                               *  Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ArduinoLinkedvariable model module.
 * @module model/ArduinoLinkedvariable
 * @version 2.0.0
 */
var ArduinoLinkedvariable = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoLinkedvariable</code>.
   * ArduinoLinkedvariable media type (default view)
   * @alias module:model/ArduinoLinkedvariable
   * @param id {String} The id of the linked variable
   * @param name {String} The name of the variable
   * @param permission {String} The permission of the linked variable
   * @param thingId {String} The id of the related thing
   * @param thingName {String} The name of the related thing
   * @param type {String} The type of the variable
   * @param variableName {String} The name of the variable in the code
   */
  function ArduinoLinkedvariable(id, name, permission, thingId, thingName, type, variableName) {
    _classCallCheck(this, ArduinoLinkedvariable);
    ArduinoLinkedvariable.initialize(this, id, name, permission, thingId, thingName, type, variableName);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ArduinoLinkedvariable, null, [{
    key: "initialize",
    value: function initialize(obj, id, name, permission, thingId, thingName, type, variableName) {
      obj['id'] = id;
      obj['name'] = name;
      obj['permission'] = permission;
      obj['thing_id'] = thingId;
      obj['thing_name'] = thingName;
      obj['type'] = type;
      obj['variable_name'] = variableName;
    }

    /**
     * Constructs a <code>ArduinoLinkedvariable</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoLinkedvariable} obj Optional instance to populate.
     * @return {module:model/ArduinoLinkedvariable} The populated <code>ArduinoLinkedvariable</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoLinkedvariable();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('last_value')) {
          obj['last_value'] = _ApiClient["default"].convertToType(data['last_value'], Object);
        }
        if (data.hasOwnProperty('last_value_updated_at')) {
          obj['last_value_updated_at'] = _ApiClient["default"].convertToType(data['last_value_updated_at'], 'Date');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('permission')) {
          obj['permission'] = _ApiClient["default"].convertToType(data['permission'], 'String');
        }
        if (data.hasOwnProperty('thing_id')) {
          obj['thing_id'] = _ApiClient["default"].convertToType(data['thing_id'], 'String');
        }
        if (data.hasOwnProperty('thing_name')) {
          obj['thing_name'] = _ApiClient["default"].convertToType(data['thing_name'], 'String');
        }
        if (data.hasOwnProperty('thing_timezone')) {
          obj['thing_timezone'] = _ArduinoTimezone["default"].constructFromObject(data['thing_timezone']);
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('variable_name')) {
          obj['variable_name'] = _ApiClient["default"].convertToType(data['variable_name'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoLinkedvariable</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoLinkedvariable</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoLinkedvariable.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data[property]) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['permission'] && !(typeof data['permission'] === 'string' || data['permission'] instanceof String)) {
        throw new Error("Expected the field `permission` to be a primitive type in the JSON string but got " + data['permission']);
      }
      // ensure the json data is a string
      if (data['thing_id'] && !(typeof data['thing_id'] === 'string' || data['thing_id'] instanceof String)) {
        throw new Error("Expected the field `thing_id` to be a primitive type in the JSON string but got " + data['thing_id']);
      }
      // ensure the json data is a string
      if (data['thing_name'] && !(typeof data['thing_name'] === 'string' || data['thing_name'] instanceof String)) {
        throw new Error("Expected the field `thing_name` to be a primitive type in the JSON string but got " + data['thing_name']);
      }
      // validate the optional field `thing_timezone`
      if (data['thing_timezone']) {
        // data not null
        _ArduinoTimezone["default"].validateJSON(data['thing_timezone']);
      }
      // ensure the json data is a string
      if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
        throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
      }
      // ensure the json data is a string
      if (data['variable_name'] && !(typeof data['variable_name'] === 'string' || data['variable_name'] instanceof String)) {
        throw new Error("Expected the field `variable_name` to be a primitive type in the JSON string but got " + data['variable_name']);
      }
      return true;
    }
  }]);
  return ArduinoLinkedvariable;
}();
ArduinoLinkedvariable.RequiredProperties = ["id", "name", "permission", "thing_id", "thing_name", "type", "variable_name"];

/**
 * The id of the linked variable
 * @member {String} id
 */
ArduinoLinkedvariable.prototype['id'] = undefined;

/**
 * Last value of the linked property
 * @member {Object} last_value
 */
ArduinoLinkedvariable.prototype['last_value'] = undefined;

/**
 * Update date of the last value
 * @member {Date} last_value_updated_at
 */
ArduinoLinkedvariable.prototype['last_value_updated_at'] = undefined;

/**
 * The name of the variable
 * @member {String} name
 */
ArduinoLinkedvariable.prototype['name'] = undefined;

/**
 * The permission of the linked variable
 * @member {String} permission
 */
ArduinoLinkedvariable.prototype['permission'] = undefined;

/**
 * The id of the related thing
 * @member {String} thing_id
 */
ArduinoLinkedvariable.prototype['thing_id'] = undefined;

/**
 * The name of the related thing
 * @member {String} thing_name
 */
ArduinoLinkedvariable.prototype['thing_name'] = undefined;

/**
 * @member {module:model/ArduinoTimezone} thing_timezone
 */
ArduinoLinkedvariable.prototype['thing_timezone'] = undefined;

/**
 * The type of the variable
 * @member {String} type
 */
ArduinoLinkedvariable.prototype['type'] = undefined;

/**
 * The name of the variable in the code
 * @member {String} variable_name
 */
ArduinoLinkedvariable.prototype['variable_name'] = undefined;
var _default = exports["default"] = ArduinoLinkedvariable;
