// @arduino/arduino-iot-client/dist/model/BatchQueryResponseSeriesMediaV1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Iot API
                                                                                                                                                                                                                                                                                                                                                                                               * Collection of all public API endpoints.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The BatchQueryResponseSeriesMediaV1 model module.
 * @module model/BatchQueryResponseSeriesMediaV1
 * @version 0.0.1
 */
var BatchQueryResponseSeriesMediaV1 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BatchQueryResponseSeriesMediaV1</code>.
   * @alias module:model/BatchQueryResponseSeriesMediaV1
   * @param aggr {String} Aggregation type
   * @param expression {String} Query
   * @param metric {String} Metric name
   */
  function BatchQueryResponseSeriesMediaV1(aggr, expression, metric) {
    _classCallCheck(this, BatchQueryResponseSeriesMediaV1);
    BatchQueryResponseSeriesMediaV1.initialize(this, aggr, expression, metric);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(BatchQueryResponseSeriesMediaV1, null, [{
    key: "initialize",
    value: function initialize(obj, aggr, expression, metric) {
      obj['aggr'] = aggr;
      obj['expression'] = expression;
      obj['metric'] = metric;
    }

    /**
     * Constructs a <code>BatchQueryResponseSeriesMediaV1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BatchQueryResponseSeriesMediaV1} obj Optional instance to populate.
     * @return {module:model/BatchQueryResponseSeriesMediaV1} The populated <code>BatchQueryResponseSeriesMediaV1</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BatchQueryResponseSeriesMediaV1();
        if (data.hasOwnProperty('aggr')) {
          obj['aggr'] = _ApiClient["default"].convertToType(data['aggr'], 'String');
        }
        if (data.hasOwnProperty('expression')) {
          obj['expression'] = _ApiClient["default"].convertToType(data['expression'], 'String');
        }
        if (data.hasOwnProperty('metric')) {
          obj['metric'] = _ApiClient["default"].convertToType(data['metric'], 'String');
        }
      }
      return obj;
    }
  }]);
  return BatchQueryResponseSeriesMediaV1;
}();
/**
 * Aggregation type
 * @member {String} aggr
 */
BatchQueryResponseSeriesMediaV1.prototype['aggr'] = undefined;

/**
 * Query
 * @member {String} expression
 */
BatchQueryResponseSeriesMediaV1.prototype['expression'] = undefined;

/**
 * Metric name
 * @member {String} metric
 */
BatchQueryResponseSeriesMediaV1.prototype['metric'] = undefined;
var _default = exports["default"] = BatchQueryResponseSeriesMediaV1;
