// @arduino/arduino-iot-client/dist/model/ThingSketch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Arduino IoT Cloud API
                                                                                                                                                                                                                                                                                                                                                                                               *  Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0
                                                                                                                                                                                                                                                                                                                                                                                               * 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ThingSketch model module.
 * @module model/ThingSketch
 * @version 2.0.0
 */
var ThingSketch = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ThingSketch</code>.
   * ThingSketchPayload describes a sketch of a thing
   * @alias module:model/ThingSketch
   */
  function ThingSketch() {
    _classCallCheck(this, ThingSketch);
    ThingSketch.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ThingSketch, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ThingSketch</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ThingSketch} obj Optional instance to populate.
     * @return {module:model/ThingSketch} The populated <code>ThingSketch</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ThingSketch();
        if (data.hasOwnProperty('sketch_version')) {
          obj['sketch_version'] = _ApiClient["default"].convertToType(data['sketch_version'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ThingSketch</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ThingSketch</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['sketch_version'] && !(typeof data['sketch_version'] === 'string' || data['sketch_version'] instanceof String)) {
        throw new Error("Expected the field `sketch_version` to be a primitive type in the JSON string but got " + data['sketch_version']);
      }
      return true;
    }
  }]);
  return ThingSketch;
}();
/**
 * The autogenerated sketch version
 * @member {module:model/ThingSketch.SketchVersionEnum} sketch_version
 */
ThingSketch.prototype['sketch_version'] = undefined;

/**
 * Allowed values for the <code>sketch_version</code> property.
 * @enum {String}
 * @readonly
 */
ThingSketch['SketchVersionEnum'] = {
  /**
   * value: "v1"
   * @const
   */
  "v1": "v1",
  /**
   * value: "v2"
   * @const
   */
  "v2": "v2"
};
var _default = exports["default"] = ThingSketch;
